src/tests/PhysicsTest.cpp

```diff
@@ -41,9 +41,10 @@ class PhysicsManagerTest : public testing::Test {
   void SetUp() override {
     // set up a default simulation config to initialize MM
     auto cfg = esp::sim::SimulatorConfiguration{};
+    cfg.createRenderer = false;
     metadataMediator_ = MetadataMediator::create(cfg);
     resourceManager_ = std::make_unique<ResourceManager>(metadataMediator_);
-    context_ = esp::gfx::WindowlessContext::create_unique(0);
```

Comment:
Wait, if you change this, then we lose the protection from the unit test when the context needs to be created. 

In your case, you need to create a brand new test function to test when the Renderer is not created.

Reply:
Doing this shows that you don't need a context in order to do physics when `createRenderer==false`, which is good to know. Maybe I can modify the tests to try both with and without context?

I already added a non-physics test with `createRenderer=false` in SimTest.cpp by the way.

Reply:
Sorry for late reply here. I want to see this change stay as-is.

I actually love that we're now guaranteed not to use OpenGL in our Physics unit tests. Conceptually, physics tests shouldn't rely on rendering. Let's imagine a future day where we've completely decoupled physics and rendering code--it'll be nice that these tests just port over directly, without having to be fixed up. By never creating a context here, we're guaranteed that our physics test code is "clean"--not accidentally relying on some rendering feature.

>we lose the protection from the unit test when the context needs to be created.
If you have in mind a unit test that involves physics and rendering, the best place for that is SimTest. Simulator is where we integrate physics and rendering. You can add objects, make them interact with physics, then call draw() or whatever and check the result.

Reply:
This has nothing to do with the visualization verification. We do that in the SimTest. No problem.

My opinion is that by doing both (with or without renderer) in the physicsTest, we have solid proof that physics simulation and the rendering are totally orthogonal to each other. Both tests should pass.

Reply:
Do we have a test in the SimTest that creates a GL context and enabled physics? No, we do not have.
We used to count on this physicsTest here until this change.

Reply:
>My opinion is that by doing both (with or without gl context) in the physicsTest, we have solid proof that physics simulation and the rendering are totally orthogonal to each other. Both tests should pass.
Ok, fair enough. We'll try to find a way to run both versions of the physics tests, hopefully without duplicating a lot of code.

Reply:
Yeah, I hear you. 

Too bad we are using the google test, not the corrade test (used in the SimTest).
Corrade test supports instanced tests. See [here](https://github.com/facebookresearch/habitat-sim/blob/master/src/tests/SimTest.cpp#L157).

This diff is to add a new feature which can enable/disable the rendering. I saw Lawrence added tests in the SimTest. So I assume this feature is well covered. 

We also know the current solution is a temporary solution as we will refactor this code in the near future. So I am totally fine if we do not touch the PhysicsTest.cpp at all in this PR. I think it is the easiest way to unblock it.

Are you comfortable with this solution?

Reply:
I've made a commit to run PhysicsTest both with and without the flag set.